src/main/scala/codacy/plugins/test/multiple/MultipleTests.scala

```diff
@@ -0,0 +1,89 @@
+
+object MultipleTests extends ITest with CustomMatchers {
+
+  val opt = "multiple"
+
+  def run(docsDirectory: File, dockerImage: DockerImage, optArgs: Seq[String]): Boolean = {
+    debug(s"Running MultipleTests:")
+    val multipleTestsDirectory = docsDirectory / DockerHelpers.multipleTestsDirectoryName
+    val languages = findLanguages(multipleTestsDirectory, dockerImage)
+    val dockerTool = createDockerTool(languages, dockerImage)
+    val toolDocumentation = new DockerToolDocumentation(dockerTool, new BinaryDockerHelper(useCachedDocs = false))
+    val dockerRunner = new BinaryDockerRunner[Result](dockerTool)()
+    val runner = new ToolRunner(dockerTool, toolDocumentation, dockerRunner)
+    val tools = languages.map(new Tool(runner, DockerRunner.defaultRunTimeout)(dockerTool, _))
+    val srcDir = multipleTestsDirectory / "src"
+    val resultFile = multipleTestsDirectory / "results.xml"
+    val resultFileXML = XML.loadFile(resultFile.toJava)
+    val expectedResults = CheckstyleFormatParser.parseResultsXml(resultFileXML).toSet
+    debug(s"${multipleTestsDirectory.name} should have ${expectedResults.size} results")
+    val configuration = {
+      val patternsPath = multipleTestsDirectory / "patterns.xml"
+      if (patternsPath.exists) {
+        val patternsFileXML = XML.loadFile(patternsPath.toJava)
+        val (patterns, extraValues) = CheckstyleFormatParser.parsePatternsXml(patternsFileXML)
+        if (patterns.isEmpty) FileCfg(Some(srcDir.pathAsString), extraValues)
+        else CodacyCfg(patterns, Some(srcDir.pathAsString), extraValues)
+      } else FileCfg(Some(srcDir.pathAsString), None)
+    }
+    tools.exists { tool =>
+      val res = runTool(tool, srcDir, configuration)
+      res match {
+        case Failure(e) =>
+          info("Got failure in the analysis:")
+          e.printStackTrace()
+          false
+        case Success(results) =>
+          if (results.sameElements(expectedResults)) {
+            debug(s"Got ${results.size} results.")
+            true
+          } else {
+            error("Tool results don't match expected results:")
+            error("Extra: ")
+            info(pprint.apply(results.diff(expectedResults), height = Int.MaxValue))
+            error("Missing:")
+            info(pprint.apply(expectedResults.diff(results), height = Int.MaxValue))
+            false
+          }
+      }
+    }
+
+  }
+
+  private def runTool(tool: Tool,
+                      multipleTestsDirectory: File,
+                      configuration: Configuration): Try[Set[PluginResult]] = {
+    val builder = Set.newBuilder[PluginResult]
+    for {
+      set <- tool.run(multipleTestsDirectory,
+                      multipleTestsDirectory.list.filter(_.isRegularFile).map(_.path).toSet,
+                      configuration)
+      toolResult <- set
+    } toolResult match {
+      case Issue(patternId, filename, message, level, category, location) =>
+        val line = location match {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Default case not specified.](https://app.codacy.com/app/Codacy/codacy-plugins-test/pullRequest?prid=4500290)